src/Jackalope/Version/GenericVersioningInterface.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+namespace Jackalope\Version;
+
+use Jackalope\Transport\VersioningInterface;
+
+/**
+ * This marker interface has to be implemented by classes which want to use the generic version handler provided by
```

Comment:
this is not a marker interface. it means that the bootstrap must inject the VersionHandler instance.

a marker interface would define no methods.